### ISSUE TITLE
Export caddyhttp.Server name field

### ISSUE THREAD

**Issue by jjiang-stripe** [caddyserver/caddy#5528]
Hello!

I'm writing a middleware that's used across a couple `caddyhttp.Server`s and it has some behaviour that differs based on the server. It'd be nice to be able to do something like [serverNameFromContext](https://github.com/caddyserver/caddy/blob/faf0399e80391ba5229321e2ee7d05262e4cc531/modules/caddyhttp/metrics.go#L89-L97), but unfortunately the `name` field is unexported, so I can't use it in my custom middleware :( 

Do y'all see any issues with exporting the `name` field? If not, I'm happy to make a PR to do so, but figured I'd check in first to make sure I'm not missing anything egregious.

**Comment by mholt** [comment]
Hi there!

I guess we can export it, sure. We already trust the programmer(s) to not change things that aren't intended/supposed to be changed, so we could do that for this and just document it as a read-only field ("pretty please") -- or we could have a getter method, like, `ServerName()` or `GetName()` or something like that, but I generally feel like those are clunky.

So yeah, maybe we just export it?

**Comment by jjiang-stripe** [comment]
I'm happy to follow whatever convention we have in Caddy whether that's to export it or add a getter! I actually noticed there's already another getter function in `server.go` so I went ahead and added a getter instead of exporting (but happy to change if you prefer). PR here: https://github.com/caddyserver/caddy/pull/5531

The middleware doesn't do anything particularly interesting–it just sets some headers based on the client certificate of the request and does some other general authorization checks. We have some Servers that require client authentication and some that don't, so it's nice to be able to do different checks based on the Server. The goal is eventually to require client auth on all our listeners, so it's a bit easier to keep it all in one middleware rather than duplicating code for separate ones.
